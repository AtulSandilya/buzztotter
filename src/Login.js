import React, { Component, PropTypes } from 'react';
import { Image, Text, TouchableHighlight, View } from 'react-native';

import { connect } from 'react-redux';

import { Actions } from 'react-native-router-flux';

import {colors} from './Styles';

class Login extends Component {
  static propTypes = {
    isLoggedIn: React.PropTypes.bool.isRequired,
  }

  render() {
    return(
      <View style={{flex: 1}}>
        <View style={{flex: 1, alignItems: 'center', justifyContent: 'center', backgroundColor: colors.bevPrimary}}>
          <Image
          source={require('../img/logos/main-logo-big.png')}
          />
          <Text style={{
            fontSize: 25,
            marginTop: 35,
            color: '#333',
          }}
          >
            Send a Beer to a Friend!
          </Text>
        </View>
        <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
          <View style={{flex: -1, alignItems: 'center'}}>
            <TouchableHighlight
              style={{
                backgroundColor: '#3b5998',
                padding: 15,
                borderRadius: 3,
              }}
              onPress={this.props.logInAction}
            >
              <Text style={{fontSize: 20, color: '#ffffff'}}>Import Contacts from Facebook</Text>
            </TouchableHighlight>
          </View>
          <View style={{flex: -1, alignItems: 'center', marginTop: 30}}>
            <TouchableHighlight
              style={{
                backgroundColor: '#dd4b39',
                padding: 15,
                borderRadius: 3,
              }}
              onPress={this.props.logInAction}
            >
              <Text style={{fontSize: 20, color: '#ffffff'}}>Import Contacts from Google</Text>
            </TouchableHighlight>
          </View>
          <View style={{flex: -1, alignItems: 'center', marginTop: 30, marginHorizontal: 30}}>
            <Text>
              * We only use your Contacts to send and receive Bevegrams.
              Your privacy is important to us. We promise to never share your information with anyone for any reason.
            </Text>
          </View>
        </View>
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    isLoggedIn: state.login.isLoggedIn,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    logInAction: () => {
      dispatch({type: 'LOG_IN'});
      Actions.mainScene();
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);
