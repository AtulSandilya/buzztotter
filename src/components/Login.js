import React, { Component, PropTypes } from 'react';
import { Image, Text, TouchableHighlight, View } from 'react-native';

import {globalColors} from './GlobalStyles';

const Login = ({isLoggedIn, logInAction}) => (
  <View style={{flex: 1}}>
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center', backgroundColor: globalColors.bevPrimary}}>
      <Image
      source={require('../../img/logos/main-logo-big.png')}
      />
      <Text style={{
        fontSize: 25,
        marginTop: 35,
        color: '#333',
      }}
      >
        Send a Beer to a Friend!
      </Text>
    </View>
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <View style={{flex: -1, alignItems: 'center'}}>
        <TouchableHighlight
          style={{
            backgroundColor: '#3b5998',
            padding: 15,
            borderRadius: 3,
          }}
          onPress={logInAction}
        >
          <Text style={{fontSize: 20, color: '#ffffff'}}>Import Contacts from Facebook</Text>
        </TouchableHighlight>
      </View>
      <View style={{flex: -1, alignItems: 'center', marginTop: 30}}>
        <TouchableHighlight
          style={{
            backgroundColor: '#dd4b39',
            padding: 15,
            borderRadius: 3,
          }}
          onPress={logInAction}
        >
          <Text style={{fontSize: 20, color: '#ffffff'}}>Import Contacts from Google</Text>
        </TouchableHighlight>
      </View>
      <View style={{flex: -1, alignItems: 'center', marginTop: 30, marginHorizontal: 30}}>
        <Text>
          * We only use your Contacts to send and receive Bevegrams.
          Your privacy is important to us. We promise to never share your information with anyone for any reason.
        </Text>
      </View>
    </View>
  </View>
)

Login.propTypes = {
  isLoggedIn: React.PropTypes.bool.isRequired,
}

export default Login;
