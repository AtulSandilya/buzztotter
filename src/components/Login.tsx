import * as React from "react";
import { Image, TouchableHighlight, View } from "react-native";

import FacebookButton from "./FacebookButton";

import { WindowHeight, WindowWidth } from "../ReactNativeUtilities";
import theme from "../theme";
import BevText from "./BevText";

export interface LoginProps {
  loginInProgress: boolean;
  goToTermsAndConditions: () => void;
  requestLogin: () => void;
}

/* tslint:disable:no-magic-numbers */
const Login: React.StatelessComponent<LoginProps> = ({
  loginInProgress,
  goToTermsAndConditions,
  requestLogin,
}) => (
  <View
    style={{
      flex: 1,
    }}
  >
    <View
      style={{
        alignItems: "center",
        backgroundColor: theme.colors.bevPrimary,
        flex: 2,
        justifyContent: "center",
      }}
    >
      <Image
        source={require("../../img/logos/logo-on-brown-big.png")}
        style={{
          height: WindowHeight * 0.3,
          width: WindowWidth * 0.8,
        }}
        resizeMode="contain"
      />
      <BevText
        color={theme.colors.white}
        size={"extraLarge"}
        textStyle={{
          marginTop: theme.padding.extraLarge,
          textAlign: "center",
        }}
      >
        Sending & Receiving Drinks Made Easy!
      </BevText>
    </View>
    <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
      <View style={{ flex: -1, alignItems: "center" }}>
        <FacebookButton
          text="Login with Facebook"
          size="largeNormal"
          onPress={requestLogin}
          showActivityIndicator={loginInProgress}
        />
      </View>
      <TouchableHighlight
        onPress={goToTermsAndConditions}
        underlayColor="transparent"
      >
        <View
          style={{
            alignItems: "center",
            flex: -1,
            marginHorizontal: theme.padding.large,
            marginTop: theme.padding.small,
          }}
        >
          <BevText size="smallNormal">
            {"By logging in, you agree to our"}
          </BevText>
          <BevText size="smallNormal" underline={true}>
            {"Terms & Privacy Policy"}
          </BevText>
        </View>
      </TouchableHighlight>
    </View>
  </View>
);
export default Login;
