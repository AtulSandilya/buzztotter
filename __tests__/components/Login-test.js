import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../build/components/Login';

it('renders correctly', () => {
  const tree = renderer.create(
    <Login
     loginInProgress={false}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();

  it('fails without props', () => {
    expect(() => {
      renderer.create(
        <Login/>
      ).toJSON();
    }).toThrow();
  });

});
