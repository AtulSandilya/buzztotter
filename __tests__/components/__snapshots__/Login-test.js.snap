// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#8ED0BA",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 400.2,
          "width": 600,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      color="#ffffff"
      disabled={false}
      ellipsizeMode="tail"
      size="extraLarge"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#ffffff",
          "fontFamily": undefined,
          "fontSize": 33.75,
          "fontWeight": "normal",
          "letterSpacing": undefined,
          "lineHeight": 51.25781249999999,
          "marginTop": 24.75,
          "textAlign": "center",
          "textAlignVertical": "center",
          "textShadowColor": "rgba(0, 0, 0, 0.25)",
          "textShadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "textShadowRadius": 2,
        }
      }
      textStyle={
        Object {
          "marginTop": 24.75,
          "textAlign": "center",
        }
      }
    >
      Sending & Receiving Drinks Made Easy!
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": -1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "elevation": 3,
              "margin": 4.888888888888888,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.18,
              "shadowRadius": 2,
            },
            Object {},
            Object {
              "borderRadius": 3,
            },
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "backgroundColor": "#3b5988",
                "borderColor": undefined,
                "borderRadius": 3,
                "borderWidth": 0,
                "flex": -1,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": -1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "paddingHorizontal": 11,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#ffffff",
                      "fontSize": 18.371173070873834,
                      "lineHeight": 18.371173070873834,
                      "textShadowColor": "rgba(0, 0, 0, 0.25)",
                      "textShadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "textShadowRadius": 2,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                ïˆ°
              </Text>
            </View>
            <Text
              accessible={true}
              allCaps={false}
              allowFontScaling={true}
              color="#ffffff"
              disabled={false}
              ellipsizeMode="tail"
              fontWeight="bold"
              isCondensed={true}
              showTextShadow={true}
              size="largeNormal"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 18.371173070873834,
                  "fontWeight": "bold",
                  "letterSpacing": undefined,
                  "lineHeight": 25.312499999999996,
                  "paddingLeft": 0,
                  "paddingRight": 11,
                  "paddingVertical": 7.333333333333333,
                  "textAlignVertical": "center",
                  "textShadowColor": "rgba(0, 0, 0, 0.25)",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": 0,
                  },
                  "textShadowRadius": 2,
                  "top": 0,
                }
              }
              textStyle={
                Object {
                  "paddingLeft": 0,
                  "paddingRight": 11,
                  "paddingVertical": 7.333333333333333,
                  "top": 0,
                }
              }
            >
              Login with Facebook
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": -1,
            "marginHorizontal": 16.5,
            "marginTop": 7.333333333333333,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          size="smallNormal"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#333333",
              "fontFamily": undefined,
              "fontSize": 12.24744871391589,
              "fontWeight": "normal",
              "letterSpacing": undefined,
              "lineHeight": 14.999999999999998,
              "textAlignVertical": "center",
            }
          }
        >
          By logging in, you agree to our
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          size="smallNormal"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#333333",
              "fontFamily": undefined,
              "fontSize": 12.24744871391589,
              "fontWeight": "normal",
              "letterSpacing": undefined,
              "lineHeight": 14.999999999999998,
              "textAlignVertical": "center",
              "textDecorationLine": "underline",
            }
          }
          underline={true}
        >
          Terms & Privacy Policy
        </Text>
      </View>
    </View>
  </View>
</View>
`;
